scheduler/scheduler_test.go

```
@@ -0,0 +1,155 @@
+package scheduler
+
+import (
+	"bytes"
+	"log/slog"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func TestScheduler(t *testing.T) {
+	t.Run("test type", func(t *testing.T) {
+		scheduler1, _ := NewTaskSchedule(5, 40)
+		got := reflect.TypeOf(scheduler1)
+		expected := reflect.TypeOf(&TaskSchedule{})
+		if got != expected {
+			t.Errorf("got %v; expected %v", got, expected)
+		}
+	})
+	t.Run("test type error", func(t *testing.T) {
+		scheduler1, _ := NewTaskSchedule(5, 40)
```

@Sagi-levi why are we ignoring the errors? 
It points that we're might not testing the real logic of the scheduler and specifically of `NewTaskSchedule`.

---

Added an error handling to the tests were needed.